app/containers/MessageBox/index.tsx

```diff
@@ -496,7 +496,11 @@ class MessageBox extends Component<IMessageBoxProps, IMessageBoxState> {
 		const msg = this.text;
 		const { start, end } = this.selection;
 		const cursor = Math.max(start, end);
-		const regexp = /([a-z0-9._-]+)$/im;
+		let regexp = /([a-z0-9._-]+)$/im;
+		if (trackingType === MENTIONS_TRACKING_TYPE_USERS) {
+			// Selects query text after '@' for all languages while mentioning a user
+			regexp = /([^@p{L}]+)$/im;
```

Comment:
You should reuse this regexp on your tests instead of repeating yourself.
You can create a small function inside `Messagebox` with it and a separate file to describe those tests.

Reply:
Made some changes @diegolmello . Created a function to get Regexp and generate its tests. Also added the tests for other mentions like emoji,room and commands.
Please review .